src/test/ui/loops/loop-no-implicit-break.rs

```diff
@@ -0,0 +1,27 @@
+fn main() {
+    let a: i8 = loop {
+        1 //~ ERROR mismatched types
+    };
+
+    let b: i8 = loop {
+        break 1;
+    };
+}
+
+fn foo() -> i8 {
+    let a: i8 = loop {
+        1 //~ ERROR mismatched types
+    };
+
+    let b: i8 = loop {
+        break 1;
+    };
+
+    loop {
+        1 //~ ERROR mismatched types
+    }
+
+    loop {
+        return 1;
+    }
```

Comment:
Can you add another case of `loop { 1 }` as `foo`'s tail expression? That could suggest either `return` or `break`, right?

Reply:
Done. Yes, I thought it best to output no more than one of the suggestions, so that case is suggesting `return` as before. Note that this is also the case with the third error, hence this new test case may seem a bit redundant.